### PR TITLE
Add NFS & Azure Files support

### LINKED ISSUE #27: Allow choice of NFS over Gluster
The /moodle directory is shared out to frontend VMs through the replicated Gluster cluster for high availability as well. However, we are hearing that the Gluster performance is not very good and asked about other options, like copying the static content over to the local file system. I think we should at least try to allow the choice of NFS over Gluster in the template. We might use one dedicated VM for the NFS host, but also could consider utilizing the jumpbox VM. Of course we should also consider the cron job scalability on the jumbbox as well, in that case.

### PR DESCRIPTION
**Pull request by hosungsmsft**
Added NFS & Azure Files support for file server type.

Azure Files option is disabled due to its very slow performance. Will evaluate its perf later when Azure Files Premium becomes available.

There are other changed params in this PR:
- Changed DB SKU default to 100 DTU, standard, 125GB
- Increased https proxy timeout a lot higher (default 1 min too short for Azure Files)

This fixes #27, though this NFS support introduces a single-point-of-failure (this NFS is not highly available). Will need to document this clearly and recommend NFS only for non-production deployments.

### CODE REVIEW

File: azuredeploy.json [hunk 1]
```diff
@@ -193,7 +205,7 @@
                 400,
                 800
             ],
-            "defaultValue": 50,
+            "defaultValue": 100,
```
Comment (rgardler) [inline azuredeploy.json]: This and subsequent changes in DB SKU increase the cost of an initial deployment. I assume the changes are based on results from load testing and we will, eventually, provide T-Shirt sizing (#36) so these changes are probably OK. However, does they really belong in this PR? Do they have any bearing on the addition of NFS?

File: nested/controllerconfig.json [hunk 2]
```diff
@@ -8,9 +8,9 @@
             },
             "type": "object"
         },
-        "blobStorageAccountKey": {
+        "storageAccountKey": {
             "metadata": {
-                "description": "Key used to create a storage container"
+                "description": "Key used to create a storage container and Azure Files share if azurefiles is specified as fileServerType"
```
Comment (rgardler) [inline nested/controllerconfig.json]: Should probably remove the reference to Azure Files in the description. Lets leave it as an undocumented feature given uncertainty of performance.

File: nested/webvmss.json [hunk 3]
```diff
@@ -7,6 +7,12 @@
                 "description": "Common Moodle values"
             },
             "type": "object"
+        },
+        "storageAccountKey": {
+            "metadata": {
+                "description": "Key used to create the Azure Files share in case azurefiles is specified as fileServerType"
```
Comment (rgardler) [inline nested/webvmss.json]: Should probably remove Azure Files reference from description.

File: azuredeploy.json (hunk 1 above)
Comment (hosungsmsft) [reply azuredeploy.json]: Reverting this before merging. I don't think these have any bearing on NFS addition, but I happened to notice slow performance with the default DB sizing (Basic 50), and somewhat inadvertently added this change in this PR, which I agree is not a good place for this change.

File: nested/controllerconfig.json (hunk 2 above)
Comment (hosungsmsft) [reply nested/controllerconfig.json]: Removed

File: nested/webvmss.json (hunk 3 above)
Comment (hosungsmsft) [reply nested/webvmss.json]: Rephrased

**Pull request merged**